script/run-comp-match

```diff
@@ -16,20 +17,85 @@ sys.path.insert(1, str(REPO_ROOT / 'modules'))
 import controller_utils  # noqa:E402 # isort:skip
 from controller_utils import NUM_ZONES  # isort:skip
 
+NUM_CONTROLLERS = NUM_ZONES + 2
+
 
 def get_zone_path(zone_id: int) -> Path:
     robot_file: Path = controller_utils.get_zone_robot_file_path(zone_id)
     return robot_file.parent
 
 
+def construct_match_data(args: argparse.Namespace) -> controller_utils.MatchData:
+    tlas = [
+        tla if tla != '-' else None
+        for tla in args.tla
+    ]
+
+    return controller_utils.MatchData(
+        args.match_num,
+        tlas,
+        args.duration,
+        recording_config=None,  # use the default
+    )
+
+
+def prepare_match(archives_dir: Path, match_data: controller_utils.MatchData) -> None:
+    controller_utils.MODE_FILE.write_text('comp\n')
+    controller_utils.record_match_data(match_data)
+
+    for zone_id, tla in enumerate(match_data.teams):
+        zone_path = get_zone_path(zone_id)
+
+        if zone_path.exists():
+            shutil.rmtree(zone_path)
+
+        if tla is None:
+            # no team in this zone
+            continue
+
+        zone_path.mkdir()
+        with ZipFile(f'{archives_dir / tla}.zip') as zipfile:
+            zipfile.extractall(zone_path)
+
+
+def run_match() -> None:
+    subprocess.check_call([
+        'webots',
+        '--batch',
+        '--stdout',
+        '--stderr',
+        '--minimize',
+        str(REPO_ROOT / 'worlds' / 'Arena.wbt'),
+    ])
+
+
+def collate_logs(archives_dir: Path, match_data: controller_utils.MatchData) -> None:
+    """
+    Copy the teams' logs into directories next to their original code archives.
+    """
+
+    for zone_id, tla in enumerate(match_data.teams):
+        if tla is None:
+            # no team in this zone
+            continue
+
+        log_filname = controller_utils.get_robot_log_filename(zone_id)
+        log_path = get_zone_path(zone_id) / log_filname
+
+        team_dir = archives_dir / tla
+        team_dir.mkdir()
```

Comment:
```suggestion
        team_dir.mkdir(exist_ok=True)
```
Since after the first match for each team this folder will exist and we aren't catching the FileExistsError

Reply:
Huh, for some reason I'd thought that this was the default. Seems not.